main.go

```diff
@@ -119,43 +137,69 @@ func main() {
 			if vmnicsValue, exists := conf.Collect["vmnics"]; exists {
 				collectvmnics = vmnicsValue
 			}
-		} else {
+		} else if !healthOnly {
 			log.Errorf("Section '%s' not found in config file", section)
 			return
 		}
 
-		log.Infof("Host: %s", *nutanixURL)
+		registry := prometheus.NewRegistry()
 
-		nutanixAPI := nutanix.NewNutanix(*nutanixURL, *nutanixUser, *nutanixPassword, maxParallelReq)
+		// always expose health per section (use host URL as key when available, else section)
+		healthSectionKey := section
+		if ok && len(conf.Host) > 0 {
+			healthSectionKey = conf.Host
```

Comment:
Why are we setting host as sectionKey here? It may not be always same.. check the configuration details